backend/dashboard/use_cases.py

```diff
@@ -1,128 +1,124 @@
-    return calculate_historical_green_transactions(user_transactions, esg_data)[0]
-    
-
-def top_5_companies(user_id) -> dict:
-    """
-    Returns in dict format:  { 'Company Name' : str, 'ESG Score' : int, 'Amount Spent' : int }
-    """
-    user_transactions = get_table_from_firebase('Users')[user_id]['transactions']
-    esg_data = get_table_from_firebase('esg')
-    return find_most_purchased_companies(user_transactions, esg_data)
-
-
-def total_co2_score(user_id) -> int:
-    """
-    Returns CO2 score for the past year.
-    """
-    user_transactions = get_table_from_firebase('Users')[user_id]['transactions']
-    esg_data = get_table_from_firebase('esg')
-    monthly_scores = [score for score in calculate_historical_scores(user_transactions, esg_data)
-                        if score is not None]
-    return int(sum(monthly_scores) / len(monthly_scores))
-
-
-def this_month_co2_score(user_id) -> int:
-    """
-    Returns CO2 score for this month.
-    """
-    user_transactions = get_table_from_firebase('Users')[user_id]['transactions']
-    esg_data = get_table_from_firebase('esg')
-    return calculate_historical_scores(user_transactions, esg_data)[0]
-
-
-def company_tiers(user_id) -> list[int]:
-    """
-    Returns list of length 4, where the first index is the number of companies in the highest tier.
-    """
-    user_transactions = get_table_from_firebase('Users')[user_id]['transactions']
-    esg_data = get_table_from_firebase('esg')
-    return find_companies_in_each_tier(user_transactions, esg_data)
-
-
-def co2_score_change(user_id) -> int:
-    """
-    Returns the difference between last month and this month's CO2 score.
-    """
-    user_transactions = get_table_from_firebase('Users')[user_id]['transactions']
-    esg_data = get_table_from_firebase('esg')
-    monthly_scores = calculate_historical_scores(user_transactions, esg_data)
-
-    if monthly_scores[0] is None or monthly_scores[1] is None:
-        return 0
-    return int(monthly_scores[0] - monthly_scores[1])
-    
-
-def green_transaction_change(user_id) -> int:
-    """
-    Returns the difference between last month and this month's # of green transactions.
-    """
-    user_transactions = get_table_from_firebase('Users')[user_id]['transactions']
-    esg_data = get_table_from_firebase('esg')
-    monthly_green_transactions = calculate_historical_green_transactions(user_transactions, esg_data)
-    if monthly_green_transactions[0] is None or monthly_green_transactions[1] is None:
-        return 0
-    return int(monthly_green_transactions[0] - monthly_green_transactions[1])
+class DashboardUseCases:
+    """DashboardUseCases class is responsible for handling the business logic of the dashboard.
+    """
+    def __init__(self, calculations: Calculations, data_access: AbstractDataAccess):
+        self.calculations = calculations
+        self.data_access = data_access
+    def past_12_month_names() -> list[str]:
```

Comment:
Can the self parameter also be added to the functions in this file?

Reply:
Yes, that's my bad I overlooked that